.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs

```diff
@@ -560,6 +560,26 @@ private List<Token> ExtractRelativeDurationDate(string text, DateObject referenc
                 if (match.Success)
                 {
                     ret.AddRange(AgoLaterUtil.ExtractorDurationWithBeforeAndAfter(text, er, ret, Config.UtilityConfiguration));
+
+                    // Check for combined patterns Duration + Date, e.g. '3 days before Monday', '4 weeks after January 15th'

```

Comment:
Could this perhaps be moved to the new merged method or it doesn't make sense?

Reply:
which merged method do you mean?

Reply:
MergeAllResults in PR #2305.

Reply:
I see, but MergeAllResults is analogous to MergeAllTokens which only merges overlapping extractions. For me, this seems more appropriate in the current method ExtractRelativeDurationDate which already deals with similar patterns like "3 days from today".